src/refactorings/convert-ternary-to-if-else/convert-ternary-to-if-else.ts

```diff
@@ -19,32 +19,31 @@ async function convertTernaryToIfElse(
   await editor.write(updatedCode.code);
 }
 
-function hasTernaryToConvert(ast: t.AST, selection: Selection): boolean {
-  let result = false;
-
-  t.traverseAST(ast, {
+function hasTernaryToConvertVisitorFactory(
+  selection: Selection,
+  onMatch: (path: t.NodePath<any>) => void
+): t.Visitor {
+  return {
     ConditionalExpression(path) {
       const { parentPath } = path;
       if (!selection.isInsidePath(path)) return;
 
       if (t.isReturnStatement(parentPath.node)) {
-        result = true;
+        onMatch(path);
       }
 
       if (t.isAssignmentExpression(parentPath.node)) {
-        result = true;
+        onMatch(path);
       }
 
       if (
         t.isVariableDeclarator(parentPath.node) &&
         t.isVariableDeclaration(parentPath.parent)
       ) {
-        result = true;
+        onMatch(path);
```

Comment:
Oh I see, this one will require more thoughts indeed.

Reply:
Certainly due to the fact that some refactorings were not yet converted to the `createVisitor()` pattern. We should refactor that first to get these follow the same (simple) implementation.